packages/wrangler/src/__tests__/publish.test.ts

```diff
@@ -116,6 +121,127 @@ describe("publish", () => {
       `);
       expect(std.err).toMatchInlineSnapshot(`""`);
     });
+
+    describe("non-TTY", () => {
+      const ENV_COPY = process.env;
+
+      afterEach(() => {
+        process.env = ENV_COPY;
+      });
+
+      it("should not throw an error in non-TTY if 'CLOUDFLARE_API_TOKEN' & 'account_id' are in scope", async () => {
+        process.env = {
+          CLOUDFLARE_API_TOKEN: "123456789",
+        };
+        setIsTTY(false);
+        writeWranglerToml({
+          account_id: "some-account-id",
+        });
+        writeWorkerSource();
+        mockSubDomainRequest();
+        mockUploadWorkerRequest();
+        mockOAuthServerCallback();
+
+        await runWrangler("publish index.js");
+
+        expect(std.out).toMatchInlineSnapshot(`
+          "Uploaded test-name (TIMINGS)
+          Published test-name (TIMINGS)
+            test-name.test-sub-domain.workers.dev"
+        `);
+        expect(std.err).toMatchInlineSnapshot(`""`);
+      });
+
+      it("should not throw an error if 'CLOUDFLARE_ACCOUNT_ID' & 'CLOUDFLARE_API_TOKEN' are in scope", async () => {
+        process.env = {
+          CLOUDFLARE_API_TOKEN: "hunter2",
+          CLOUDFLARE_ACCOUNT_ID: "some-account-id",
+        };
+        setIsTTY(false);
+        writeWranglerToml();
+        writeWorkerSource();
+        mockSubDomainRequest();
+        mockUploadWorkerRequest();
+        mockOAuthServerCallback();
+        mockGetMemberships({
+          success: true,
+          result: [],
+        });
+
+        await runWrangler("publish index.js");
+
+        expect(std.out).toMatchInlineSnapshot(`
+          "Uploaded test-name (TIMINGS)
+          Published test-name (TIMINGS)
+            test-name.test-sub-domain.workers.dev"
+        `);
+        expect(std.err).toMatchInlineSnapshot(`""`);
+      });
+
+      it("should throw an error in non-TTY if 'account_id' & 'CLOUDFLARE_ACCOUNT_ID' is missing", async () => {
+        setIsTTY(false);
+        process.env = {
+          CLOUDFLARE_API_TOKEN: "hunter2",
+          CLOUDFLARE_ACCOUNT_ID: undefined,
+        };
+        writeWranglerToml({
+          account_id: undefined,
+        });
+        writeWorkerSource();
+        mockSubDomainRequest();
+        mockUploadWorkerRequest();
+        mockOAuthServerCallback();
+        mockGetMemberships({
+          success: true,
+          result: [
+            { id: "IG-88", account: { id: "1701", name: "enterprise" } },
+            { id: "R2-D2", account: { id: "nx01", name: "enterprise-nx" } },
+          ],
+        });
+
+        await expect(runWrangler("publish index.js")).rejects
```

Comment:
The test doesn't match the title of this test

Reply:
To clarify, in non-tty, we should throw an error if an account id isn't available via the toml or an env var, and it shouldn't make the fetch to get memberships.

Reply:
Account ID was already being handled in `getAccountId()`
The environment is checked, including "legacy" account ID here:
```ts
  const accountIdFromEnv = getCloudflareAccountIdFromEnv();
  if (accountIdFromEnv) {
    return accountIdFromEnv;
  }
  ```
> To clarify, in non-tty, we should throw an error if an account id isn't available via the toml or an env var, and it shouldn't make the fetch to get memberships.

From my conversation with @petebacondarwin I would say there may be differing in opinion on how this should be handled, and I would rather have a conversation with the three of us then.

Reply:
That's fair, I don't feel super strongly about it right now. What does the error/message look like if the api token is invalid?

Reply:
```ts
  throw new Error(
        "More than one account available but unable to select one in non-interactive mode.\n" +
          `Please set the appropriate \`account_id\` in your \`wrangler.toml\` file.\n` +
          `Available accounts are ("<name>" - "<id>"):\n` +
          responseJSON.result
            .map((x) => `  "${x.account.name}" - "${x.account.id}")`)
            .join("\n")
      );
```

Reply:
Created a fork, ran Wrangler in CI on itself, trying to publish a fake file created inline, this is the output (without the change is TBD yet)
<img width="1387" alt="Screen Shot 2022-05-03 at 5 16 21 PM" src="https://user-images.githubusercontent.com/27247160/166575707-427910e6-74fc-4f4e-a3ed-2e75957a6ffd.png">

Reply:
Looks like the error doesn't actually stop the login flow? Hence the timeout message

Reply:
Did you test this with an invalid token, or an empty token? The error implies no token at all?

Reply:
could you try this with a garbage token and see what the error is?